packages/react/src/auto/polaris/inputs/PolarisAutoEnumInput.tsx

```diff
@@ -0,0 +1,149 @@
+  const {
+    allowMultiple,
+    allowOther,
+    onSelectionChange,
+    selectedOptions,
+    allOptions,
+    filteredOptions,
+    searchQuery,
+    label,
+    metadata,
+    isError,
+    errorMessage,
+  } = useEnumInputController({ field: fieldApiIdentifier, control });
+  const { value: searchValue, setValue: setSearchValue } = searchQuery;
+
+  const selectedTagsElement = useMemo(
+    () =>
+      allowMultiple && selectedOptions.length > 0 ? (
+        <InlineStack gap="150">
+          {selectedOptions.map((tag) => (
+            <Tag key={`option-${tag}`} onRemove={() => onSelectionChange(tag)}>
+              {tag}
+            </Tag>
+          ))}
+        </InlineStack>
+      ) : null,
+    [allowMultiple, onSelectionChange, selectedOptions]
+  );
+
+  const optionItemElement = useMemo(() => {
+    if (allOptions.length === 0) return null;
+
+    const formatOptionText = (option: string) => {
+      const trimValue = searchValue.trim().toLocaleLowerCase();
+      const matchIndex = option.toLocaleLowerCase().indexOf(trimValue);
+
+      if (!searchValue || matchIndex === -1) return option;
+
+      const start = option.slice(0, matchIndex);
+      const highlight = option.slice(matchIndex, matchIndex + trimValue.length);
+      const end = option.slice(matchIndex + trimValue.length, option.length);
+
+      return (
+        <p>
+          {start}
+          <Text fontWeight="bold" as="span">
+            {highlight}
+          </Text>
+          {end}
+        </p>
+      );
+    };
+
+    return filteredOptions.map((option) => {
+      return (
+        <Listbox.Option key={option} value={option} selected={selectedOptions.includes(option)} accessibilityLabel={option}>
+          <Listbox.TextOption selected={selectedOptions.includes(option)}>{formatOptionText(option)}</Listbox.TextOption>
+        </Listbox.Option>
+      );
+    });
+  }, [allOptions.length, filteredOptions, searchValue, selectedOptions]);
+
+  const addExtraOptionElement = useMemo(() => {
+    if (!allowOther) return null;
+
+    const noResults = searchValue && !allOptions.includes(searchValue);
+    return noResults ? <Listbox.Action value={searchValue}>{`Add "${searchValue}"`}</Listbox.Action> : null;
+  }, [allOptions, allowOther, searchValue]);
+
+  const emptyStateElement = useMemo(() => {
+    if (allowOther || (optionItemElement && optionItemElement.length > 0) || !searchValue) return null;
+    return <EmptySearchResult title="" description={`No  found matching "${searchValue}"`} />;
+  }, [allowOther, optionItemElement, searchValue]);
+
+  useEffect(() => {
+    if (allowMultiple) {
+      setSearchValue("");
+      return;
+    }
+
+    if (selectedOptions.length > 0) {
+      // Set the search value to show the selected option in the input field
+      setSearchValue(selectedOptions[0]);
+    } else {
+      setSearchValue("");
+    }
+  }, [allowMultiple, selectedOptions, setSearchValue]);
+
+  const listBoxElement = useMemo(
```

Comment:
In general, I'm not sure that these helpers that create an element need to be memoized. They don't seem to do much complicated/expensive computation, and I think there is a risk of creating subtle bugs if we forget to add something to the cache invalidation array.

I think these can just be regular helpers that generate markup, like: 

```
let listBoxElement = null;
if(optionItemElement || addExtraOpetionElement || emptyStateElement) {

 listBoxElement = ( 
  <Listbox ..
  )
  ```